co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstance.java

```diff
@@ -167,27 +234,40 @@ public void bind() throws IOException {
      * @throws IOException If there is an issue with the underlying socket object or
      *                     methods
      */
-    public void sendMsgs(byte[] data) throws IOException {
-        if (rxMsgsSocket == null) {
-            throw new IllegalStateException("Attempted to send data before opening socket");
-        }
-        DatagramPacket packet = new DatagramPacket(data, data.length, targetAddress, rxMessagePort);
-        rxMsgsSocket.send(packet);
-
+    public void sendV2xMsg(byte[] data) throws IOException {
+        sendPacket(data, rxMessagePort);
     }
 
     /**
-     * Sends time sync data to the Infrastructure Device communications interface configured at construction time.
+     * Sends time sync data to the Infrastrucutre Instance Time Sync port.
+     * 
+     * @param data The binary data to transmit
+     * @throws IOException If there is an issue with the underlying socket object or methods
+     */
+    public void sendTimeSyncMsg(byte[] data) throws IOException {
+        sendPacket(data, timeSyncPort);
+    }
+    /**
+     * Sends time sync data to the Infrastrucutre Instance Simulated Interaction port.
+     * 
      * @param data The binary data to transmit
      * @throws IOException If there is an issue with the underlying socket object or methods
      */
-    public void sendTimeSyncMsgs(byte[] data) throws IOException {
-        if (rxMsgsSocket == null) {
+    public void sendInteraction(byte[] data) throws IOException {
+        sendPacket(data, simulatedInteractionPort);
+    }
+    /**
+     * Method to send byte data to specified port using the infrastructure instance Datagramsocket.
+     * 
+     * @param data byte data to send using Datagramsocket.
+     * @param port in integer format to send Datagram to.
+     * @throws IOException
+     */
+    public void sendPacket(byte[] data, int port) throws IOException {
+         if (socket == null) {
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/net/SocketException.html Do we know the specific socket exception at this point? Maybe checking whether the socket is ready when we initialize the infrastructure instance? so it guarantees that the socket is ready for use here?

Reply:
The only precondition for the socket is that it is not null/initialized by the bind method, which we already check. Additionally I am not sure editing the UDPSockets we use for CDASim Infrastructure Ambassador really falls under the scope of this story.